src/main/java/com/yelp/nrtsearch/server/luceneserver/QueryNodeMapper.java

```diff
@@ -660,8 +664,109 @@ private SpanQuery getSpanQuery(
       default:
         throw new IllegalArgumentException("Unsupported Span Query: " + protoSpanQuery);
     }
   }
+
+  private SpanMultiTermQueryWrapper getSpanMultiTermQueryWrapper(
+      com.yelp.nrtsearch.server.grpc.SpanMultiTermQueryWrapper protoSpanMultiTermQueryWrapper,
+      IndexState state) {
+
+    com.yelp.nrtsearch.server.grpc.SpanMultiTermQueryWrapper.WrappedQueryCase wrappedQueryCase =
+        protoSpanMultiTermQueryWrapper.getWrappedQueryCase();
+
+    switch (wrappedQueryCase) {
+      case WILDCARDQUERY:
+        WildcardQuery wildcardQuery =
+            new WildcardQuery(
+                new Term(
+                    protoSpanMultiTermQueryWrapper.getWildcardQuery().getTerm().getField(),
+                    protoSpanMultiTermQueryWrapper.getWildcardQuery().getTerm().getText()));
+        return new SpanMultiTermQueryWrapper<>((MultiTermQuery) wildcardQuery);
+      case FUZZYQUERY:
+        FuzzyQuery fuzzyQuery = getFuzzyQuery(protoSpanMultiTermQueryWrapper);
+        return new SpanMultiTermQueryWrapper<>((MultiTermQuery) fuzzyQuery);
+      case PREFIXQUERY:
+        Query prefixQuery = getPrefixQuery(protoSpanMultiTermQueryWrapper.getPrefixQuery(), state);
+        if (prefixQuery instanceof MultiTermQuery) {
+          return new SpanMultiTermQueryWrapper<>((MultiTermQuery) prefixQuery);
+        }
+      case REGEXPQUERY:
+        RegexpQuery regexpQuery = getRegexpQuery(protoSpanMultiTermQueryWrapper);
+        if (regexpQuery instanceof MultiTermQuery) {
+          return new SpanMultiTermQueryWrapper<>((MultiTermQuery) regexpQuery);
+        }
+      default:
+        throw new IllegalArgumentException(
+            "Unsupported Span Multi Query Term Wrapper: " + protoSpanMultiTermQueryWrapper);
+    }
+  }
+
+  private static FuzzyQuery getFuzzyQuery(
+      com.yelp.nrtsearch.server.grpc.SpanMultiTermQueryWrapper protoSpanMultiTermQueryWrapper) {
+    com.yelp.nrtsearch.server.grpc.FuzzyQuery protoFuzzyQuery =
+        protoSpanMultiTermQueryWrapper.getFuzzyQuery();
+    Term term = new Term(protoFuzzyQuery.getTerm().getField(), protoFuzzyQuery.getTerm().getText());
+
+    int maxEdits = protoFuzzyQuery.getMaxEdits();
+    int prefixLength = protoFuzzyQuery.getPrefixLength();
+    int maxExpansions = protoFuzzyQuery.getMaxExpansions();
+    boolean transpositions = protoFuzzyQuery.getTranspositions();
+
+    if (maxEdits != FuzzyQuery.defaultMaxEdits
+        && prefixLength != FuzzyQuery.defaultPrefixLength
+        && maxExpansions != FuzzyQuery.defaultMaxExpansions
+        && !transpositions) {
+      return new FuzzyQuery(term, maxEdits, prefixLength, maxExpansions, transpositions);
+    } else if (maxEdits != FuzzyQuery.defaultMaxEdits
+        && prefixLength != FuzzyQuery.defaultPrefixLength) {
+      return new FuzzyQuery(term, maxEdits, prefixLength);
+    } else if (maxEdits != FuzzyQuery.defaultMaxEdits) {
+      return new FuzzyQuery(term, maxEdits);
+    }
+    return new FuzzyQuery(term);
```

Comment:
To make this a bit easier to understand, it may be better to set all the parameter to either the specified or default values, then invoke a single constructor will everything.